docs/SPIR-V.rst

```diff
@@ -3282,6 +3282,140 @@ Interface Variables
 ``IncomingCallableData{NV/KHR}``        First argument of entry for Callable stage
 =================================       ===========================================================
 
+Ray Query
+
+Ray Query is subfeature of the directx raytracing and belongs to the directx raytracing spec 1.1 (DXR 1.1)
+DirectX add RayQuery object type and its member TraceRayInline() to do the TraceRay() that doesn't
+use any seperate ray-tracing shader stages.
+Shaders can instantiate RayQuery objects as local variables, the RayQuery object acts as a state
+machine for ray query. The shader interacts with the RayQuery object’s methods to advance the
+query through an acceleration structure and query traversal information
+
+| Refer to following pages for details:
+| https://microsoft.github.io/DirectX-Specs/d3d/Raytracing.html
+
+A flow chart for a simple ray query process
+
+          +------------------------------+
+          |   RayQuery<RAY_FLAG_NONE> q  |
+          +------------------------------+
+                         |
+                         V
+          +------------------------------+
+          |      q.TraceRayInline()      |
+          +------------------------------+
+                  |               — — — — — — — — — — — — —
+                  |              |                         |
+                  |              |              +------------------------+
+                  |              |              | Your intersection code |
+                  |              |              +------------------------+
+                  |              |                         ^
+                  V              V                         |
+          +------------------------------+      +---------------------+
+          |  q.Proceed() // AS traversal |      |  q.CandidateType()  |
+          +------------------------------+      +---------------------+
+               |                   |                       ^
+           No  |                   | Yes                   |
+               |                   |_ _ _ _ _ _ _ _ _ _ _ _|
+               V
+         +------------------------------+
+         |     q.CommittedStatus()      |
+         +------------------------------+
+                       |
+                       V
+        +----------------------------------+
+        | Your Intersection/shader code    |
+        +----------------------------------+
+
+
+.. Example flow chart code:: hlsl
+
+void main() {
```

Comment:
need to have `.. code:: hlsl` followed by an empty line before this line.
See line 2975 as an example.